clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2253,44 +2262,49 @@ public void testPollAuthenticationFailure(GroupProtocol groupProtocol) throws In
+        try (final KafkaConsumer<String, String> consumer = consumerWithPendingAuthenticationError(groupProtocol)) {
+            assertThrows(AuthenticationException.class, () -> consumer.offsetsForTimes(singletonMap(tp0, 0L)));
+        }
     }
 
     // TODO: this test triggers a bug with the CONSUMER group protocol implementation.
     //       The bug will be investigated and fixed so this test can use both group protocols.
     @ParameterizedTest
     @EnumSource(value = GroupProtocol.class, names = "CLASSIC")
     public void testCommitSyncAuthenticationFailure(GroupProtocol groupProtocol) {
-        final KafkaConsumer<String, String> consumer = consumerWithPendingAuthenticationError(groupProtocol);
-        Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
-        offsets.put(tp0, new OffsetAndMetadata(10L));
-        assertThrows(AuthenticationException.class, () -> consumer.commitSync(offsets));
+        try (final KafkaConsumer<String, String> consumer = consumerWithPendingAuthenticationError(groupProtocol)) {
+            Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
+            offsets.put(tp0, new OffsetAndMetadata(10L));
+            assertThrows(AuthenticationException.class, () -> consumer.commitSync(offsets));
+        }
     }
 
     // TODO: this test triggers a bug with the CONSUMER group protocol implementation.
     //       The bug will be investigated and fixed so this test can use both group protocols.
     @ParameterizedTest
     @EnumSource(value = GroupProtocol.class, names = "CLASSIC")
     public void testCommittedAuthenticationFailure(GroupProtocol groupProtocol) {
-        final KafkaConsumer<String, String> consumer = consumerWithPendingAuthenticationError(groupProtocol);
-        assertThrows(AuthenticationException.class, () -> consumer.committed(Collections.singleton(tp0)).get(tp0));
+        try (final KafkaConsumer<String, String> consumer = consumerWithPendingAuthenticationError(groupProtocol)) {
+            assertThrows(AuthenticationException.class, () -> consumer.committed(Collections.singleton(tp0)).get(tp0));
+        }
     }
 
     @ParameterizedTest
     @EnumSource(value = GroupProtocol.class)
     public void testMeasureCommitSyncDurationOnFailure(GroupProtocol groupProtocol) {
-        final KafkaConsumer<String, String> consumer
-            = consumerWithPendingError(groupProtocol, new MockTime(Duration.ofSeconds(1).toMillis()));
+        try (final KafkaConsumer<String, String> consumer
+            = consumerWithPendingError(groupProtocol, new MockTime(Duration.ofSeconds(1).toMillis()))) {
 
-        try {
-            consumer.commitSync(Collections.singletonMap(tp0, new OffsetAndMetadata(10L)));
-        } catch (final RuntimeException e) {
-        }
+            try {
+                consumer.commitSync(Collections.singletonMap(tp0, new OffsetAndMetadata(10L)));
```

Comment:
```suggestion
                consumer.commitSync(Map.of(tp0, new OffsetAndMetadata(10L)));
```

Reply:
Ack. Will apply throughout the whole class.